### PR TITLE
ROB: continue parsing dictionnary object when error is detected

### LINKED ISSUE #2866: #5 Using PdfReader causes a crash
Hi!

Another crash similar to previous ones. Pdf and stderr could be found below.

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python -m platform
Linux-5.15.0-56-generic-x86_64-with-glibc2.31

$ python -c "import pypdf;print(pypdf._debug_versions)"
pypdf==5.0.0, crypt_provider=('cryptography', '3.1'), PIL=none
```

commit 6cfa0c4fe8d8d9a144e32f120066e9de17005d04

## Code + PDF

This is a minimal, complete example that shows the issue:

```python
#! /usr/bin/env python3

import pypdf
from pypdf.errors import EmptyFileError, PdfReadError, PdfStreamError
import sys

def TestOneInput(fname):
  try:
    pdf_reader = pypdf.PdfReader(fname)
    for page_number, page in enumerate(pdf_reader.pages):
        page.extract_text()
  except (EmptyFileError, PdfReadError, PdfStreamError):
      pass

if __name__ == "__main__":
    if len(sys.argv) < 2:
        exit(1)
    TestOneInput(sys.argv[1])
```

### PoC

[crash-e108c4f677040b61e12fa9f1cfde025d704c9b0d.pdf](https://github.com/user-attachments/files/17095516/crash-e108c4f677040b61e12fa9f1cfde025d704c9b0d.pdf)

## Traceback

This is the complete stderr I see:

```
PdfReadError("Invalid Elementary Object starting with b'M' @276: b'ASGAA+Arial,Unicode MS\\n/DescendantFonts [ 56 0 R ]\\n/Encoding /Identity-H\\n/Subtyp'")
Traceback (most recent call last):
  File "/fuzz/./poc.py", line 18, in <module>
    TestOneInput(sys.argv[1])
  File "/fuzz/./poc.py", line 11, in TestOneInput
    page.extract_text()
  File "/usr/local/lib/python3.9/dist-packages/pypdf/_page.py", line 2266, in extract_text
    return self._extract_text(
  File "/usr/local/lib/python3.9/dist-packages/pypdf/_page.py", line 1761, in _extract_text
    cmaps[f] = build_char_map(f, space_width, obj)
  File "/usr/local/lib/python3.9/dist-packages/pypdf/_cmap.py", line 32, in build_char_map
    font_subtype, font_halfspace, font_encoding, font_map = build_char_map_from_dict(
  File "/usr/local/lib/python3.9/dist-packages/pypdf/_cmap.py", line 53, in build_char_map_from_dict
    font_type: str = cast(str, ft["/Subtype"])
  File "/usr/local/lib/python3.9/dist-packages/pypdf/generic/_data_structures.py", line 441, in __getitem__
    return dict.__getitem__(self, key).get_object()
KeyError: '/Subtype'
```

### PR DESCRIPTION
**Pull request by pubpub-zz**
closes #2866

### CODE REVIEW

File: pypdf/generic/_data_structures.py [hunk 1]
```diff
@@ -1443,9 +1454,13 @@ def read_object(
         else:
             return NumberObject.read_from_stream(stream)
     else:
+        pos = stream.tell()
         stream.seek(-20, 1)
+        txt = stream.read(80)
```
Comment (stefan6419846) [inline pypdf/generic/_data_structures.py:1459]: This name is misleading - this isn't really text, but rather bytes, if I am not mistaken.

File: tests/test_reader.py [hunk 2]
```diff
@@ -1657,3 +1657,49 @@ def test_comments_in_array(caplog):
     reader.stream = BytesIO(b[:1149])
     with pytest.raises(PdfStreamError):
         reader.pages[0]
+
+
+@pytest.mark.enable_socket()
+def test_space_in_names_to_continue_processing(caplog):
+    """
+    This deals with space not encoded in names inducing errors
+    Also covers case where NameObject not met for key
```
Comment (stefan6419846) [inline tests/test_reader.py:1666]:
```suggestion
    Also covers case where NameObject not met for key.
```

File: pypdf/_protocols.py [hunk 3]
```diff
@@ -59,6 +59,8 @@ def strict(self) -> bool:
 
 
 class PdfReaderProtocol(PdfCommonDocProtocol, Protocol):
+    strict: bool
```
Comment (stefan6419846) [inline pypdf/_protocols.py:62]: Why hasn't this been required before?

Comment (pubpub-zz) [reply pypdf/_protocols.py:62]: Strange... now it seems to not be required...
Will remove it

File: pypdf/generic/_data_structures.py (hunk 1 above)
Comment (pubpub-zz) [reply pypdf/generic/_data_structures.py:1459]: will change to stream_extract

**Review by stefan6419846 — APPROVED** [review]

**Pull request merged**